src/main/java/com/google/api/codegen/LanguageUtil.java

```diff
@@ -67,4 +67,8 @@ public static String lowerUnderscoreToLowerCamel(String name) {
   public static String lowerCamelToUpperCamel(String name) {
     return CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_CAMEL, name);
   }
+
+  public static String lowerCamelToLowerUnderscore(String name) {
```

Comment:
Could you change the "written  twice" in the class comment above? It seems like now CodegenContext delegates to these static methods, as it should.